### PR TITLE
Adds support for diffusion

### PR DESCRIPTION
**Pull request by dchansen**
This PR adds support for diffusion scans in MRD.
It also has a number of updates for Matlab, to allow it to use the latest additions to MRD.

### CODE REVIEW

File: libsrc/xml.cpp [hunk 1]
```diff
@@ -265,6 +265,51 @@ namespace ISMRMRD
...
+    }
+
+  static Diffusion parse_diffusion(pugi::xml_node& node) {
+      Diffusion diff{};
+      diff.bvalue = std::stof(node.child_value("bvalue"));
+
+      auto grad_node = node.child("gradientDirection");
+
+      diff.gradientDirection.rl = std::stof(grad_node.child_value("rl"));
+      diff.gradientDirection.rl = std::stof(grad_node.child_value("ap"));
```
Comment (kspaceKelvin) [inline libsrc/xml.cpp:293]: Typo in this and the below line :)

**Comment by hansenms** [comment]
This looks good to me. I am a little reluctant to hit approve since I have not tested it. I just looked through. @ojosephs would you be able to take a look? @kspaceKelvin any other comments?

**Comment by ojosephs** [comment]
I can't see any issues, but some (most!) of the coding is seriously above my level! I'm happy to be talked through it, perhaps with Kelvin?

**Comment by dchansen** [comment]
@kspaceKelvin Ready for you

File: libsrc/ismrmrd.cpp [hunk 2]
```diff
@@ -1216,24 +1216,24 @@ template <> EXPORTISMRMRD inline ISMRMRD_DataTypes get_data_type<complex_double_
...
 // Images
-template EXPORTISMRMRD class Image<uint16_t>;
-template EXPORTISMRMRD class Image<int16_t>;
-template EXPORTISMRMRD class Image<uint32_t>;
-template EXPORTISMRMRD class Image<int32_t>;
-template EXPORTISMRMRD class Image<float>;
-template EXPORTISMRMRD class Image<double>;
-template EXPORTISMRMRD class Image<complex_float_t>;
-template EXPORTISMRMRD class Image<complex_double_t>;
+template class EXPORTISMRMRD Image<uint16_t>;
```
Comment (kspaceKelvin) [inline libsrc/ismrmrd.cpp:1219]: Is this an intentional bugfix?

File: schema/ismrmrd_example_extended.xml [hunk 3]
```diff
@@ -376,6 +376,32 @@
...
+    </diffusion>
+    <diffusion>
+      <gradientDirection>
+        <ap>0</ap>
+        <fh>0</fh>
+        <rl>1</rl>
+      </gradientDirection>
+      <bvalue>4.0</bvalue>
+    </diffusion>
+	  <diffusionScheme>bipolar</diffusionScheme>
```
Comment (kspaceKelvin) [inline schema/ismrmrd_example_extended.xml:404]: Can you please fix the spacing on this so it renders properly online?

**Review by kspaceKelvin — COMMENTED** [review]
Looks good otherwise!

File: libsrc/ismrmrd.cpp (hunk 2 above)
Comment (dchansen) [reply libsrc/ismrmrd.cpp:1219]: Yes. Won't compile in VS 2022 otherwise.

File: schema/ismrmrd_example_extended.xml (hunk 3 above)
Comment (dchansen) [reply schema/ismrmrd_example_extended.xml]: Sure. Should be fixed now.

**Comment by dchansen** [comment]
@kspaceKelvin I think I addressed all the issues we discussed. Is it ok to merge?

**Review by hansenms — APPROVED** [review]

**Pull request merged**